skademlia/protocol.go

```diff
@@ -144,6 +144,15 @@ func (p Protocol) Client(info noise.Info, ctx context.Context, authority string,
 	return conn, nil
 }
 
+func addressMatches(bind string, subject string) bool {
+	bindHost, bindPort, _ := net.SplitHostPort(bind)
```

Comment:
We should handle errors and return false immediately as either the binding or subject IP address is in-parseable.